### PR TITLE
feat: render 404 or Staking view based on poolUrls array.

### PR DESCRIPTION
**Pull request by Tulun**
1. Renders staking view for matching pool url.
2. Renders 404 if that value is not contained in string[]

Screen Shot 2022-08-29 at 11 15 16 AM
Screen Shot 2022-08-29 at 11 15 09 AM

### CODE REVIEW

File: src/Routes.tsx [hunk 1]
```diff
@@ -81,7 +82,20 @@ const Routes: React.FC = () => {
...
+            const poolIdFound = poolUrls.includes(
+              match.params.poolId.toLowerCase()
+            );
+            if (poolIdFound) {
+              return <Staking />;
+            } else {
+              return <NotFound custom404Message="Pool not found." />;
+            }
+          }}
+        />
```
[the remark this answers is not in the document]
Comment (Tulun) [reply src/Routes.tsx]: Sure, I've made a generic function.

File: src/views/Pool.tsx [hunk 2]
```diff
@@ -36,6 +36,7 @@ const Pool: FC = () => {
   };
   const config = getConfig();
   const tokenList = config.getTokenList(chainId);
+  console.log("TL", tokenList);
```
[the remark this answers is not in the document]
Comment (Tulun) [reply src/views/Pool.tsx]: done.

File: src/Routes.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tulun) [reply src/Routes.tsx]: IMO this is good for now.

The nice thing is the NotFound view doesn't run a lot of hooks and stuff that won't have a good value anyway -- in addition, we don't have to add additional checks for bad values.

If Tim has a different idea for a 404 though we can go that route later.

File: src/utils/constants.ts [hunk 3]
```diff
@@ -705,3 +705,18 @@ export const usdcLpCushion = process.env.REACT_APP_USDC_LP_CUSHION || "0";
 export const wethLpCushion = process.env.REACT_APP_WETH_LP_CUSHION || "0";
 export const wbtcLpCushion = process.env.REACT_APP_WBTC_LP_CUSHION || "0";
 export const daiLpCushion = process.env.REACT_APP_DAI_LP_CUSHION || "0";
+
+export const poolUrls = [
```
Comment (daywiss) [inline src/utils/constants.ts:709]: this should be dynamically generated by the apps configuration.  there is a function to get list of supported tokens: https://github.com/across-protocol/frontend-v2/blob/master/src/utils/config.ts#L139 and you can get the config with this function: https://github.com/across-protocol/frontend-v2/blob/master/src/utils/config.ts#L200

File: src/Routes.tsx [hunk 4]
```diff
@@ -81,7 +83,22 @@ const Routes: React.FC = () => {
         <Route exact path="/about" component={About} />
         <Route exact path="/rewards" component={Rewards} />
         <Route exact path="/airdrop" component={Claim} />
-        <Route exact path="/rewards/staking/:poolId" component={Staking} />
+        <Route
+          exact
+          path="/rewards/staking/:poolId"
+          render={({ match }) => {
+            const poolIdFound = stringValueInArray(
+              match.params.poolId,
```
Comment (daywiss) [inline src/Routes.tsx:91]: you might want to toLower it just in case

**Review by daywiss — COMMENTED** [review]
looks good, just want to make sure we dont have extra configuration maintenence when we can avoid it

**Pull request merged**